core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -397,6 +399,28 @@ class SpecificUnitAutomation{
         else UnitAutomation().explore(unit, unit.getDistanceToTiles())
     }
 
+    fun automateGeneral(unit: MapUnit){
+        //try to follow nearby units. Do not garrison in city if possible
+        val militantToCompany = unit.civInfo.getCivUnits()
+                .firstOrNull { val tile = it.currentTile
```

Comment:
Instead of this being firstOrNull, I would filter first, then order by closeness, so the great general always tries to go to the closest unit that fills the requirements

Reply:
I don't know much about Kotlin. Is firstOrNull more performance efficient than filtering all units? What I am actually trying to do, is finding first militant that the general can reach within one turn. Any suggestions?

Reply:
FirstOrNull is more efficient when you have a heavy boolean check, since once one element satisfies the check, it doesn't continue to check the others
Ah, if you're looking for "within one turn" then that's different, you'll want to use unit.algs().distanceToTiles() to start with
Instead of iterating over all civ units, you can start by iterating over all tiles in range of one turn to find a close military unit, that way the number of things you need to check is much smaller